Source/NSSplitViewController.m

```diff
@@ -0,0 +1,208 @@
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import <Foundation/NSArchiver.h>
+
+#import "AppKit/NSSplitView.h"
+#import "AppKit/NSSplitViewController.h"
+#import "AppKit/NSSplitViewItem.h"
+#import "AppKit/NSView.h"
+#import "NSViewPrivate.h"
+
+#import "GSFastEnumeration.h"
+
+@implementation NSSplitViewController
+// return splitview...
+- (NSSplitView *) splitView
+{
+  return (NSSplitView *)[self view];
+}
+
+- (void) setSplitView: (NSSplitView *)splitView
+{
+  [self setView: splitView];
+  [splitView setDelegate: self];
+}
+
+- (NSSplitViewItem *) splitViewItemForViewController: (NSViewController *)vc
+{
+  FOR_IN (NSSplitViewItem*, svi, _splitViewItems)
+    if ([svi viewController] == vc)
+      {
+        return svi;
+      }
+  END_FOR_IN (_splitViewItems);
+  return nil;
+}
+
+- (CGFloat) minimumThicknessForInlineSidebars
+{
+  return _minimumThicknessForInlineSidebars;
+}
+
+- (void) setMinimumThicknessForInlineSidebars: (CGFloat)value
+{
+  _minimumThicknessForInlineSidebars = value;
+}
+  
+// manage splitview items...
+- (NSArray *) splitViewItems
+{
+  return _splitViewItems;
+}
+
+- (void) setSplitViewItems: (NSArray *)items
+{
+  NSMutableArray *mutableItems = [items mutableCopy];
+  ASSIGN(_splitViewItems, mutableItems);
+}
+
+- (void) addSplitViewItem: (NSSplitViewItem *)item
+{
+  [self insertSplitViewItem: item atIndex: [_splitViewItems count]];
+}
+
+- (void) insertSplitViewItem: (NSSplitViewItem *)item atIndex: (NSInteger)index
+{
+  NSSplitView *sv = [self splitView];
+  NSViewController *vc = [item viewController];
+
+  if (vc != nil)
+    {
+      NSView *v = [vc view];
+      if (v != nil)
+        {
+          [sv _insertSubview: v atIndex: index];
+        }
+    }       
+  [_splitViewItems insertObject: item atIndex: index];     
+}
+
+- (void) removeSplitViewItem: (NSSplitViewItem *)item
+{
+  NSViewController *vc = [item viewController];
+  if (vc != nil)
+    {
+      NSView *v = [vc view];
+      if (v != nil)
+        {
+          [[self splitView] removeSubview: v];  
+        }
+    }
+  [_splitViewItems removeObject: item];
+}
+
+- (void) dealloc
+{
+  RELEASE(_splitViewItems);
+  [super dealloc];
+}
+
+// instance methods...
+- (NSRect) splitView: (NSSplitView *)splitView additionalEffectiveRectOfDividerAtIndex: (NSInteger)dividerIndex
+{
+  return [splitView frame];
```

Comment:
All these methods are not really implemented.

Reply:
These are really for subclasses to implement.  They are the delegate methods for the NSSplitView.